bot/exts/backend/sync/_syncers.py

```diff
@@ -161,17 +163,32 @@ async def _get_diff(self, guild: Guild) -> _Diff:
 
         for db_user in db_users.values():
             guild_user = guild_users.get(db_user.id)
+
             if guild_user is not None:
                 if db_user != guild_user:
```

Comment:
This is redundant and inefficient since you'll compare each attribute again when this is true.